bgpd/bgp_route.c

```diff
@@ -4105,7 +4105,11 @@ int bgp_update(struct peer *peer, const struct prefix *p, uint32_t addpath_id,
 		if (aspath_get_last_as(attr->aspath) == bgp->as)
 			do_loop_check = 0;
 
-	if (CHECK_FLAG(peer->af_flags[afi][safi], PEER_FLAG_REFLECTOR_CLIENT))
+	if (CHECK_FLAG(peer->af_flags[afi][safi], PEER_FLAG_REFLECTOR_CLIENT) ||
+	    (safi == SAFI_UNICAST && !peer->afc[afi][safi] &&
+	     peer->afc[afi][SAFI_LABELED_UNICAST] &&
+	     CHECK_FLAG(peer->af_flags[afi][SAFI_LABELED_UNICAST],
+			PEER_FLAG_REFLECTOR_CLIENT)))
```

Comment:
you can't add this much unconditional change without ... a comment explaining exactly what these additional terms mean.

Reply:
Commit log was updated with a comment.

Reply:
That's good, from the commit-log pov, but it's really what I asked for? I'd like to see a comment here in the code, where this very particular decision is being taken. Otherwise, someone who is trying to follow the logic bears the burden of doing a git blame, walking through the code, and searching the git history in order to locate the comment.